### ISSUE TITLE
convolve_fft does not work well with some 3d kernels, claiming a lot of memory is needed

### ISSUE THREAD

**Issue by astrofrog** [astropy/astropy#10047]
This works fine:

```python
In [1]: import numpy as np                                                      
In [2]: image = np.random.random((768, 768))                                    
In [3]: kernel = np.random.random((41, 41))                                     
In [4]: from astropy.convolution import convolve_fft                            
In [5]: _ = convolve_fft(image, kernel)                   
```        

But this does not but should be equivalent:

```python       
In [6]: image = np.random.random((1, 768, 768))                                 
In [7]: kernel = np.random.random((1, 41, 41))                                  
In [8]: _ = convolve_fft(image, kernel)                                         
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-24dbf28e9436> in <module>
----> 1 _ = convolve_fft(image, kernel)
~/Code/astropy/astropy/nddata/decorators.py in wrapper(data, *args, **kwargs)
    244                                   AstropyUserWarning)
    245 
--> 246             result = func(data, *args, **kwargs)
    247 
    248             if unpack and repack:
~/Code/astropy/astropy/convolution/convolve.py in convolve_fft(array, kernel, boundary, fill_value, nan_treatment, normalize_kernel, normalization_zero_tol, preserve_nan, mask, crop, return_fft, fft_pad, psf_pad, min_wt, allow_huge, fftn, ifftn, complex_dtype)
    686         raise ValueError("Size Error: Arrays will be {}.  Use "
    687                          "allow_huge=True to override this exception."
--> 688                          .format(human_file_size(array_size_C)))
    689 
    690     # For future reference, this can be used to predict "almost exactly"
ValueError: Size Error: Arrays will be  17G.  Use allow_huge=True to override this exception.
```

**Comment by pllim** [comment]
This is due to

https://github.com/astropy/astropy/blob/24dc50ac020b25ac395ee74679a90dabc3acb1f0/astropy/convolution/convolve.py#L673

that increased the new array shape from `(1024, 1024)` to `(1024, 1024, 1024)` purely based on `array.ndim` even if one of the dimensions is size 1.

The code has been pretty much unchanged since #660 by @keflavich .

**Comment by keflavich** [comment]
yeah this looks like a big oversight; we should be squeezing before doing any of this

**Comment by pllim** [comment]
But should the output be 3d or 2d?

**Comment by keflavich** [comment]
3d; original ndim should be preserved.  Literal `.squeeze()` isn't the solution, that would be too easy...

**Comment by dhomeier** [comment]
This would be solved by #11533; do you think a specific test for 3D input is warranted?

**Comment by neutrinoceros** [comment]
This was indeed fixed and the original example code doesn't raise an exception anymore.
Tests for 3D convolution are still missing but I'm planning to add some in #15896 
I'd recommend closing this.
